### PR TITLE
Use new screening interface in test_screening

### LINKED ISSUE #938: convert unit_test/test_screening to new style screening
we are still doing:
```
jscr_init++;                                                                                                                                                                                                                          
add_screening_factor(jscr_init, zion[ic12],aion[ic12],zion[ih1],aion[ih1]);
```

### PR DESCRIPTION
**Pull request by maxpkatz**
Closes #938

### CODE REVIEW

**Comment by zingale** [comment]
there are roundoff diffs:
http://groot.astro.sunysb.edu/Microphysics/test-suite/gfortran/2022-07-04-008/index.html

is that expected?

**Comment by maxpkatz** [comment]
`calculate_screening_factor` uses gcem's code for evaluating exponentials, which doesn't have to be the same as whatever code is generated by the compiler for std::exp. So I don't expect it to result in exactly the same numerical screening factors.

**Comment by maxpkatz** [comment]
Which reminds me that in C++20 we could use `std::is_constant_evaluated` to have the gcem exp default back to std::exp in non-consteval contexts (https://en.cppreference.com/w/cpp/types/is_constant_evaluated).

**Comment by zingale** [comment]
yes, C++ 20 will make this all nicer.  Not sure when we can rely on that support though.

File: unit_test/test_screening/screening_util.cpp [hunk 1]
```diff
@@ -219,287 +112,217 @@ void screen_test_C(const Box& bx,
...
+    actual_screen(pstate, scn_fac, sc1a, sc1adt);
     sp(i, j, k, vars.iscn_co55_p) = sc1a;
     sp(i, j, k, vars.iscn_co55_p_dt) = sc1adt;
 
     // fe54(p,g)co55
-    jscr++;
-    screen(pstate, jscr,
-           zion[ife54], aion[ife54], zion[ih1], aion[ih1],
-           sc1a, sc1adt, sc1add);
+    scn_fac = scrn::calculate_screen_factor(zion[ife52], aion[ife54], zion[ih1], aion[ih1]);
```
[the remark this answers is not in the document]
Comment (maxpkatz) [reply unit_test/test_screening/screening_util.cpp]: fixed

**Review by zingale — APPROVED** [review]

**Pull request merged**